### PR TITLE
ECR Push Feature

### PR DESCRIPTION
**Pull request by warlordwinter**
# Dockerize Rubber Duck
#### Summery
- This feature will automate creating Docker Images for Rubber Duck using Github actions. 
- This feature will trigger when Main receives a push command.
- A future feature will be created to allow for quick and easy set up for the project. Stay tuned
- Bellow is picture of the working tests.
screenshot_of_workflowtests

### CODE REVIEW

File: Dockerfile
Comment (byubean) [inline Dockerfile:1]: delete this file

File: .github/workflows/ecr.push.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Make build_docker.sh executable
+        run: chmod +x ./build_docker.sh
+
+      - name: Run Docker build script
+        run: ./build_docker.sh
```
Comment (byubean) [inline .github/workflows/ecr.push.yml:22]: Instead of `chmod`ing the file, just do `bash build_docker.sh`

File: build_docker.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+IMAGE_TAG="latest"
+
+docker build \
+    -t ${IMAGE_NAME}:${IMAGE_TAG} \
+    -f - . <<EOF
+FROM python:3.11.9
+LABEL authors="Wiley Welch, Bryce Martin, Gordon Bean"
+COPY rubber_duck /rubber_duck
+ADD pyproject.toml /rubber_duck/pyproject.toml
+ENV OPENAI_API_KEY=${OPENAI_API_KEY}
```
Comment (byubean) [inline build_docker.sh:15]: remove this line. The user should provide this env var when running the docker

File: build_docker.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+WORKDIR /rubber_duck
+RUN pip install poetry
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction --no-ansi
+EXPOSE 8080
+CMD ["python", "/rubber_duck/discord_bot.py", "--config", "/config.json", "--log-console"]
+EOF
+
+# Tag the image for ECR
+docker tag ${IMAGE_NAME}:${IMAGE_TAG} 844825014198.dkr.ecr.us-west-2.amazonaws.com/beanlab/rubber-duck:${IMAGE_TAG}
```
Comment (byubean) [inline build_docker.sh:25]: make the ECR repo name a variable like Image name and tag name

File: rubber_duck/discord_bot.py [hunk 4]
```diff
@@ -364,6 +364,20 @@ def main(config):
...
-    config = json.loads(args.config.read_text())
+    try:
+        if args.config.is_file():
+            config = json.loads(args.config.read_text())
+        else:
+            default_config_path = Path('config.json')
+            if default_config_path.is_file():
+                config = json.loads(default_config_path.read_text())
+            else:
+                raise FileNotFoundError
```
Comment (byubean) [inline rubber_duck/discord_bot.py:375]: `raise FileNotFoundError(file)`

**Comment by warlordwinter** [comment]
@byubean Ready for round two of examinations.

**Review by byubean — APPROVED** [review]

**Pull request merged**